File: java/13514022_ITI/src/makhluk/Matahari.java
package makhluk;
/**Kelas 'Matahari'. Turunan (public) dari Bintang
 * @author Malvin J./13514044
 * {@inheritDoc}
 */
public class Matahari extends Bintang {
    /** Ctor makhluk 'Matahari'.
    * Selain atribut id, posisi, dan gambar, nilai atribut dirandom.
    * Gambar = '*'
    *
    * @param id ID
    * @param x absis awal
    * @param y ordinat awal
    * @param time waktu lahir
    *
    */
    public Matahari(final int id, final int x, final int y, final int time) {
        final double kali = 0.8;
        setArah();
        setKekuatan();
        setID(id);
        setPos(x, y);
        setDTime();
        setGambar('@');
        setStartTime(time);
        setCahaya((int) kali * super.getKekuatan());
    }
}

File: cpp/2A - NONTHREAD - STL/Matahari.cpp
/*  Nama: Sashi Novitasari/13514027
    Tanggal: 3/3/2016
    Kelas Makhluk */

#include "Shape.h"

using namespace std;

Matahari::Matahari(int id, int x, int y, int time){
    setArah();
    setKekuatan();
    ID = id;
    pos = new Point(x,y);
    setDTime();
    gambar = '@';
    starttime = time;
    cahaya = (int) 0.8*kekuatan;
}

Matahari::~Matahari(){}
